themes/theme-gmd/components/ProductCard/index.jsx

```diff
@@ -55,10 +55,10 @@ const ProductCard = ({
         )}
         {!hidePrice && (
           <Fragment>
-            <Portal name={portals.PRODUCT_ITEM_PRICE_BEFORE} props={{ productId: product.id }} />
-            <Portal name={portals.PRODUCT_ITEM_PRICE} props={{ productId: product.id }}>
+            <Portal name={portals.PRODUCT_ITEM_PRICE_BEFORE} props={{ productId: product.id, location: 'productGrid' }} />
```

Comment:
What is the value of adding the location property to the portals props?

Reply:
It's needed because PRODUCT_ITEM_PRICE(...) portal position is sometimes inconsistent. For every position except the ProductList it uses column layout. For ProductList it's row layout, which is important when one wants to add something there and style accordingly. 

So the primary goal is to make the extension aware of the real position where's it's rendered. It should be:
- productGrid 
- productCard
- productList (that's the quirky one)
- and I think we're missing ProductDetailPage here, worth checking out.

For this particular extension we need it it would render the content everytime in PRODUCT_ITEM_PRICE_AFTER except if it's productList location. For this case it would render nothing there, but in PRODUCT_ITEM_PRICE_INSIDE portal.

It's not the nicest and cleanest approach so I understand your confusion fully, but that's the quickest and most reasonable one we have right now ;)

Reply:
And still, this particular line of code is wrong. It should be adjusted to `productCard`. Please check others as well.

Reply:
For ProductDetailPage there are portals for before and after striked price, regular price, and price info inside the header. Made adjustments for productCard.